FORMS-984: Added rest webform endpoint for getting all submissions (#5)
OS2Forms/os2forms_rest_api: README.md

We use [Key auth](https://www.drupal.org/project/key_auth) for authenticating
api users.

A user can access the Webform REST API if

1. it has the “OS2Form REST API user” (`os2forms_rest_api_user`) role,

> 2. has been granted access to the formular

Reviewer: ```suggestion
2. has been granted access to the form
```

(the `sid` value is a webform submission uuid).

### Webform submissions

You can filter results based on submission time by

> adding query parameters to the url:

Reviewer: ```suggestion
adding query parameters to the URL:
```

(cf. https://www.merriam-webster.com/dictionary/URL)

| Name        | Value                | Example      |
|-------------|----------------------|--------------|

> | `starttime` | PHP DateTime formats | `yesterday`  |

Reviewer: ```suggestion
| `starttime` | [PHP Date and Time Formats](https://www.php.net/manual/en/datetime.formats.php) | `yesterday`  |
```

| `endtime`   | PHP DateTime formats | `2023-10-23` |

If left out, filtering upon the left out parameter will not be done.

> The example beneath requests all submissions after October 1. 2023.

Reviewer: ```suggestion
This example requests all submissions after on or after October 1st, 2023:
```

Request:

```sh
> curl --silent --header 'api-key: …' 'https://127.0.0.1:8000/webform_rest/some_webform_id/submissions?starttime=2023-10-01'
```

Response:

```json

> {"webform_id":"some_webform_id",
>   "starttime":"2023-10-01",
>   "submissions":{
>     "123":"https:\/\/127.0.0.1:8000\/da\/webform_rest\/some_webform_id\/submission\/44b1fe1b-ee96-481e-b941-d1219d1dcb55",
>     "124":"https:\/\/127.0.0.1:8000\/da\/webform_rest\/some_webform_id\/submission\/3652836d-3dab-4919-b880-e82cbbf3c24c"
>   }
> }

Reviewer: Consider pretty-printing this JSON value:

```suggestion
{
  "webform_id": "some_webform_id",
  "starttime": "2023-10-01",
  "submissions": {
    "123": "https://127.0.0.1:8000/da/webform_rest/some_webform_id/submission/44b1fe1b-ee96-481e-b941-d1219d1dcb55",
    "124": "https://127.0.0.1:8000/da/webform_rest/some_webform_id/submission/3652836d-3dab-4919-b880-e82cbbf3c24c"
  }
}```
